File: src/brs/web/api/http/handler/GetPeers.java
package brs.web.api.http.handler;

import brs.peer.Peer;
import brs.peer.Peers;
import brs.util.Convert;
import brs.web.api.http.ApiServlet;
import brs.web.api.http.common.LegacyDocTag;
import com.google.gson.JsonArray;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;

import javax.servlet.http.HttpServletRequest;

import static brs.web.api.http.common.Parameters.ACTIVE_PARAMETER;
import static brs.web.api.http.common.Parameters.STATE_PARAMETER;

public final class GetPeers extends ApiServlet.JsonRequestHandler {

  public static final GetPeers instance = new GetPeers();

  private GetPeers() {
    super(new LegacyDocTag[] {LegacyDocTag.INFO}, ACTIVE_PARAMETER, STATE_PARAMETER);
  }

  @Override
  protected
  JsonElement processRequest(HttpServletRequest req) {

    boolean active = "true".equalsIgnoreCase(req.getParameter(ACTIVE_PARAMETER));
    String stateValue = Convert.emptyToNull(req.getParameter(STATE_PARAMETER));

    JsonArray peers = new JsonArray();
    for (Peer peer : active ? Peers.getActivePeers() : stateValue != null ? Peers.getPeers(Peer.State.valueOf(stateValue)) : Peers.getAllPeers()) {
      peers.add(peer.getPeerAddress());
    }

    JsonObject response = new JsonObject();
    response.add("peers", peers);
    return response;
  }

}

File: src/main/java/brs/http/GetPeers.kt
package brs.http

import brs.DependencyProvider
import brs.http.common.Parameters.ACTIVE_PARAMETER
import brs.http.common.Parameters.STATE_PARAMETER
import brs.peer.Peer
import brs.util.Convert
import com.google.gson.JsonArray
import com.google.gson.JsonElement
import com.google.gson.JsonObject
import javax.servlet.http.HttpServletRequest

internal class GetPeers(private val dp: DependencyProvider) : APIServlet.JsonRequestHandler(arrayOf(APITag.INFO), ACTIVE_PARAMETER, STATE_PARAMETER) {
    override fun processRequest(request: HttpServletRequest): JsonElement {

        val active = "true".equals(request.getParameter(ACTIVE_PARAMETER), ignoreCase = true)
        val stateValue = Convert.emptyToNull(request.getParameter(STATE_PARAMETER))

        val peers = JsonArray()
        for (peer in if (active) dp.peers.activePeers else if (stateValue != null) dp.peers.getPeers(Peer.State.valueOf(stateValue)) else dp.peers.allPeers) {
            peers.add(peer.peerAddress)
        }

        val response = JsonObject()
        response.add("peers", peers)
        return response
    }
}
